cache/cache.go

```diff
@@ -289,3 +303,93 @@ func (p2 *progressWriterAt) WriteAt(p []byte, off int64) (n int, err error) {
 	p2.task.Add(n)
 	return n, errors.WithStack(err)
 }
+
+// matches: https://github.com/{OWNER}/{REPO}/releases/download/{TAG}/{ARTEFACT}
+var githubRe = regexp.MustCompile(`^https\://github.com/([^/]+)/([^/]+)/releases/download/([^/]+)/([^/]+)$`)
+
+type githubReleaseInfo struct {
+	owner, repo, tag, asset, token string
+}
+
+func getGithubReleaseInfo(uri string) (*githubReleaseInfo, bool) {
+	m := githubRe.FindStringSubmatch(uri)
+	if len(m) != 5 {
+		return nil, false
+	}
+	token := os.Getenv("HERMIT_GITHUB_TOKEN")
```

Comment:
Move this before the regex check. no point doing that work if you are just going to bail out here anyway.

Perhaps also, use the same form as the others:

```
g := &githubReleaseInfo{}
g.token = os.Getenv(...)
```